pkg/model/block.go

```diff
@@ -14,9 +11,26 @@ type Block struct {
 
 	blockID iotago.BlockID
 
-	data      []byte
-	blockOnce sync.Once
-	block     *iotago.Block
+	data       []byte
+	block      *iotago.Block
+	commitment *Commitment
+}
+
+func NewBlock(blockID iotago.BlockID, iotaBlock *iotago.Block, data []byte, api iotago.API) (*Block, error) {
```

Comment:
Does this need to be exported? You only want to go through the BlockFromBlock constructor passing an iotago.Block no?